src/app/workflow/launch-third-party/launch-third-party.component.ts

```diff
@@ -321,4 +344,48 @@ export class LaunchThirdPartyComponent extends Base implements OnChanges, OnInit
     }
     return `Export this workflow version to ${platform}.`;
   }
+
+  launchToCodespace() {
+    const correctDevcontainerPath = this.computeCorrectDevcontainerPath(this.devcontainers);
+    if (correctDevcontainerPath) {
+      this.openNewCodespaceWindow(correctDevcontainerPath);
+    } else {
+      this.displayLaunchToCodespaceDialog(this.devcontainers.length > 0)
+        .afterClosed()
+        .subscribe((shouldLaunch: boolean) => {
+          if (shouldLaunch) {
+            this.openNewCodespaceWindow(undefined);
+          }
+        });
+    }
+  }
+
+  private displayLaunchToCodespaceDialog(hasDevcontainer: boolean) {
+    return this.dialog.open(LaunchToCodespaceDialogComponent, {
+      width: bootstrap4largeModalSize,
+      data: { entry: this.workflow, hasDevcontainer: hasDevcontainer },
+    });
+  }
+
+  private computeCorrectDevcontainerPath(devcontainers: SourceFile[]): string | undefined {
+    // Return the path, relative to root, of the devcontainer with a body that contains the absolute notebook path as a substring,
+    // or undefined if there is no such devcontainer
+    return devcontainers.find((file) => file.content.includes(this.selectedVersion.workflow_path))?.absolutePath?.substring(1);
+  }
+
+  private openNewCodespaceWindow(devcontainerPath: string | undefined) {
+    let url: string =
+      this.config.GITHUB_CODESPACES_IMPORT_URL +
+      '?hide_repo_select=true' +
+      '&ref=' +
```

Comment:
Do codespaces only work against branches? 

If they also work against tags, do you still use the `ref` query parameter and/or the same format for the value?

Reply:
Yup, they work for tags, same `ref` argument.  The one caveat is that the tag name will show up as the selected "Branch" on the GitHub "Create codespace" page (it's a GitHub shortcoming, haven't found a way around it).